internal/command/clusters/uninstall.go

```diff
@@ -0,0 +1,349 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/clock"
+
+	"github.com/jetstack/jsctl/internal/command/types"
+	"github.com/jetstack/jsctl/internal/kubernetes"
+	"github.com/jetstack/jsctl/internal/kubernetes/clients"
+	"github.com/jetstack/jsctl/internal/kubernetes/status/components"
+)
+
+const (
+	hasOwnerRefInfoTemplate      = "%s/%s secret has certificate owner ref"
+	hasOwnerRefHeader            = "Secrets with Certificate owner refs were found. These Secrets will be garbage collected when Certificate CRD is uninstalled.\nYou can run 'jsctl experimental clusters cleanup secrets remove-certificate-owner-refs' command to remove the owner references."
+	upcomingRenewalInfoTemplate  = "%s/%s certificate will be renewed soon (%s)"
+	upcomingRenewalInfoHeader    = "Some certificates will be renewed soon. You might want to ensure that uninstall is completed before any renewals kick in. Or use 'cmctl renew' command to renew the certificates now."
+	upcomingExpiriesInfoTemplate = "%s/%s certificate will expire soon (%s)"
+	upcomingExpiriesHeader       = "Some certificates will expire soon. Ensure that enough time is allocated for re-installation to prevent outages. You can use 'cmctl renew' command to manually renew certs now."
+	currentIssuancesInfoTemplate = "%s/%s certificate has issuing condition set to true"
+	currentIssuancesHeader       = "Some certificates are currently being issued. You might want to ensure that issuances complete before starting to uninstall to avoid duplicate requests for certificates."
+	unreadyInfoTemplate          = "%s/%s certificate has ready condition set to false or does not have it"
+	unreadyHeader                = "Some certificates are currently not ready. You might want to fix any issues before upgrading."
+	failedInfoTemplate           = "%s/%s certificate has failed last %d issuance attempts"
+	failedInfoHeader             = "Some certificates are currently failing issuance attempts. You might want to fix any issues before uninstalling."
+	integrationHeader            = "A cert-manager integration that creates certificate requests was found in cluster. You might want to ensure that uninstalling Jetstack Secure software will not cause downtime."
+	integrationInfoTemplate      = "%s found in cluster"
```

Comment:
what is the reason for putting all these messages in consts instead of just inlining them?

Reply:
so I can use them in tests instead of inlining there too. I really don't want to figure out where some string needs to be changed in tests if in future we slightly change the naming for a few of these